pymc/tests/logprob/test_joint_logprob.py

```diff
@@ -309,3 +317,279 @@ def test_multiple_rvs_to_same_value_raises():
+    ):
+        _get_scaling({1, 2, 5}, (2, 3), 2)
+
+    # test with rvar from model graph
+    with pm.Model() as m2:
+        rv_var = pm.Uniform("a", 0.0, 1.0)
+    total_size = []
+    assert _get_scaling(total_size, shape=rv_var.shape, ndim=rv_var.ndim).eval() == 1.0
+
+
+def test_joint_logp_basic():
+    """Make sure we can compute a log-likelihood for a hierarchical model with transforms."""
+
+    with pm.Model() as m:
+        a = pm.Uniform("a", 0.0, 1.0)
+        c = pm.Normal("c")
+        b_l = c * a + 2.0
+        b = pm.Uniform("b", b_l, b_l + 1.0)
+
+    a_value_var = m.rvs_to_values[a]
+    assert m.rvs_to_transforms[a]
+
+    b_value_var = m.rvs_to_values[b]
+    assert m.rvs_to_transforms[b]
+
+    c_value_var = m.rvs_to_values[c]
+
+    (b_logp,) = joint_logp(
+        (b,),
+        rvs_to_values=m.rvs_to_values,
+        rvs_to_transforms=m.rvs_to_transforms,
+        rvs_to_total_sizes={},
+    )
+
+    # There shouldn't be any `RandomVariable`s in the resulting graph
+    assert_no_rvs(b_logp)
+
+    res_ancestors = list(walk_model((b_logp,)))
+    assert b_value_var in res_ancestors
+    assert c_value_var in res_ancestors
+    assert a_value_var in res_ancestors
+
+
+@pytest.mark.parametrize(
+    "indices, size",
+    [
+        (slice(0, 2), 5),
+        (np.r_[True, True, False, False, True], 5),
+        (np.r_[0, 1, 4], 5),
+        ((np.array([0, 1, 4]), np.array([0, 1, 4])), (5, 5)),
+    ],
+)
+def test_joint_logp_incsubtensor(indices, size):
+    """Make sure we can compute a log-likelihood for ``Y[idx] = data`` where ``Y`` is univariate."""
+
+    mu = pm.floatX(np.power(10, np.arange(np.prod(size)))).reshape(size)
+    data = mu[indices]
+    sigma = 0.001
+    rng = np.random.RandomState(232)
+    a_val = rng.normal(mu, sigma, size=size).astype(pytensor.config.floatX)
+
+    rng = pytensor.shared(rng, borrow=False)
+    a = pm.Normal.dist(mu, sigma, size=size, rng=rng)
+    a_value_var = a.type()
+    a.name = "a"
+
+    a_idx = at.set_subtensor(a[indices], data)
+
+    assert isinstance(a_idx.owner.op, (IncSubtensor, AdvancedIncSubtensor, AdvancedIncSubtensor1))
+
+    a_idx_value_var = a_idx.type()
+    a_idx_value_var.name = "a_idx_value"
+
+    a_idx_logp = joint_logp(
+        (a_idx,),
+        rvs_to_values={a_idx: a_value_var},
+        rvs_to_transforms={},
+        rvs_to_total_sizes={},
+    )
+
+    logp_vals = a_idx_logp[0].eval({a_value_var: a_val})
+
+    # The indices that were set should all have the same log-likelihood values,
+    # because the values they were set to correspond to the unique means along
+    # that dimension.  This helps us confirm that the log-likelihood is
+    # associating the assigned values with their correct parameters.
+    a_val_idx = a_val.copy()
+    a_val_idx[indices] = data
+    exp_obs_logps = sp.norm.logpdf(a_val_idx, mu, sigma)
+    np.testing.assert_almost_equal(logp_vals, exp_obs_logps)
+
+
+def test_joint_logp_subtensor():
```

Comment:
Same here

Reply:
I am sorry but I have trouble identifying where this one and the next one should go

Reply:
This subtensor test is just testing functionality that is implemented in `test_mixture`, so we can probably remove it (confirming with codecov that we are not missing lines that were covered exclusively by these)